packages/legacy/core/App/utils/contacts.ts

```diff
@@ -0,0 +1,57 @@
+import { BasicMessageRecord, ConnectionRecord, CredentialExchangeRecord, ProofExchangeRecord } from '@credo-ts/core'
+
+import { BifoldAgent } from './agent'
+
+interface ConnectionWithMessages {
+  conn: ConnectionRecord
+  msgs: (BasicMessageRecord | CredentialExchangeRecord | ProofExchangeRecord)[]
+}
+
+interface ConnectionWithLatestMessage {
+  conn: ConnectionRecord
+  latestMsg: BasicMessageRecord | CredentialExchangeRecord | ProofExchangeRecord
+}
+
+/**
+ * Function to fetch contacts (connections) in order of latest chat message without using hooks
+ * @param agent - Credo agent
+ * @returns ConnectionRecord[] sorted by most recent message
+ */
+export const fetchContactsByLatestMessage = async (agent: BifoldAgent): Promise<ConnectionRecord[]> => {
+  const connections = await agent.connections.getAll()
+  const connectionsWithMessages = await Promise.all<ConnectionWithMessages>(
```

Comment:
If any one of the promises fails all those after it will fails. Often better to use `Promise.allSettled` but I'm not sure if it's possible for any to fail in this case.

Reply:
In this case if any fail, all fail